src/org/labkey/test/components/QueryMetadataEditorPage.java

```diff
@@ -0,0 +1,91 @@
+package org.labkey.test.components;
+
+import org.labkey.test.Locator;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class QueryMetadataEditorPage extends DomainDesignerPage
+{
+    public QueryMetadataEditorPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    @Override
+    public void clickFinish()
+    {
+        scrollIntoView(elementCache().finishButton());
+        shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().finishButton()));
+        click(Locator.button(elementCache().finishButton().getText()));
+    }
+
+    public void reset()
+    {
+        scrollIntoView(newElementCache().resetButton());
+        shortWait().until(ExpectedConditions.elementToBeClickable(newElementCache().resetButton()));
+        click(Locator.button(newElementCache().resetButton().getText()));
+        click(Locator.button("Reset")); // Reset confirmation on the confirm modal
+    }
+
+    public void viewData()
+    {
+        scrollIntoView(newElementCache().viewDataButton());
+        shortWait().until(ExpectedConditions.elementToBeClickable(newElementCache().viewDataButton()));
+        click(Locator.button(newElementCache().viewDataButton().getText()));
+    }
+
+    public void editSource()
+    {
+        scrollIntoView(newElementCache().editSourceButton());
+        shortWait().until(ExpectedConditions.elementToBeClickable(newElementCache().editSourceButton()));
+        click(Locator.button(newElementCache().editSourceButton().getText()));
+    }
+
+    public void aliasField()
+    {
+        scrollIntoView(newElementCache().aliasFieldButton());
+        shortWait().until(ExpectedConditions.elementToBeClickable(newElementCache().aliasFieldButton()));
+        click(Locator.button(newElementCache().aliasFieldButton().getText()));
+        click(Locator.button("OK")); //the selected option is the first field
+    }
+
+    @Override
+    protected DomainDesignerPage.ElementCache elementCache()
+    {
+        return super.elementCache();
+    }
```

Comment:
If you cast this then you won't need to call `newElementCache` in other methods.
```suggestion
    protected QueryMetadataEditorPage.ElementCache elementCache()
    {
        return (QueryMetadataEditorPage.ElementCache) super.elementCache();
    }
```